### ISSUE TITLE
Upgrade `get-port` to 6.0.0 (or higher)

### ISSUE THREAD

**Issue by hasezoey** [nodkz/mongodb-memory-server#569]
Upgrade Dependency `get-port` to `6.0.0` or higher

This currently cannot be done, because:

- the module (`get-port`) got refactored to only use / provide ESM (ECMAScript Modules)
- jest currently (as of 27) requires a flag to activate ESM
- jest with the flag enabled **only** accepts ESM (no CJS anymore)

see [ECMAScript Modules - Jest](https://jestjs.io/docs/ecmascript-modules)

**Comment by RandomSpaceMonkey** [comment]
Hi! I'm new to open source and would like to take a jab at this.

**Comment by hasezoey** [comment]
> Hi! I'm new to open source and would like to take a jab at this.

i would not recommend it, as the original message says: `get-port` `6.0.0` or higher is ESM only, which can be conflicting with package managers and NodeJS versions, so i have postponed the upgrade until either:
- nodejs & package manager (and other librarys) properly support ESM
- the repository's packages itself get changed to use ESM only (which i am not convinced yet)

and i would like to start with providing both CJS and ESM modules (for easy upgrade, and to see if it even is possible to transition yet)

TL;DR: because `get-port` is now ESM only, it is hard to provide both CJS and ESM modules at the same time

---

Currently known problematic library:
- Jest (the testing framework) requires a flag to run ESM modules, and can **ONLY** run ESM modules in that mode
- NestJS (the webserver framework) may crash with ESM modules

**Comment by hasezoey** [comment]
update for jest 28: nothing much changed in jest for ESModules, meaning it still wont be updated

**Comment by hasezoey** [comment]
out of curiosity i tried to run this package as a ESM package, but ran into some issues, for now the main blocker is https://github.com/kulshekhar/ts-jest/issues/4127 (and maybe other issues, but i couldnt get to those yet thanks to this)

**Comment by hasezoey** [comment]
closing in favor of #579
